File: app/src/main/java/com/android/learn/mcontract/SearchContract.java
package com.android.learn.mcontract;

import com.android.base.mmodel.FeedArticleListData;

/**
 * Created by gaolei on 2018/6/18.
 */

public class SearchContract {

    public interface Presenter {


        void getFeedArticleList(String key);

        void onLoadMore(String key);

        void addCollectArticle(int position, FeedArticleListData.FeedArticleData feedArticleData);

        void cancelCollectArticle(int position, FeedArticleListData.FeedArticleData feedArticleData);
    }

    public interface View {


        void showArticleList(FeedArticleListData itemBeans);

        void showCollectArticleData(int position, FeedArticleListData.FeedArticleData feedArticleData);

        void showCancelCollectArticleData(int id, FeedArticleListData.FeedArticleData feedArticleData);
    }
}

File: app/src/main/java/com/android/learn/mcontract/SearchContract.kt
package com.android.learn.mcontract

import com.android.learn.base.mmodel.FeedArticleListData
import com.android.learn.base.mmodel.HotKeyData
import com.android.learn.base.mview.BaseView

/**
 * Created by gaolei on 2018/6/18.
 */

class SearchContract {

    interface Presenter {


        fun getFeedArticleList(key: String)

        fun onLoadMore(key: String)

        fun addCollectArticle(position: Int, feedArticleData: FeedArticleListData.FeedArticleData)

        fun cancelCollectArticle(position: Int, feedArticleData: FeedArticleListData.FeedArticleData)
    }

    interface View : BaseView {


        fun showArticleList(itemBeans: FeedArticleListData)

        fun showCollectArticleData(position: Int, feedArticleData: FeedArticleListData.FeedArticleData)

        fun showCancelCollectArticleData(id: Int, feedArticleData: FeedArticleListData.FeedArticleData)
    }
}
